src/crc32c_arm64_check.h

```diff
@@ -38,7 +38,13 @@ inline bool CanUseArm64Crc32() {
   // From 'arch/arm64/include/uapi/asm/hwcap.h' in Linux kernel source code.
   constexpr unsigned long kHWCAP_PMULL = 1 << 4;
   constexpr unsigned long kHWCAP_CRC32 = 1 << 7;
-  unsigned long hwcap = (&getauxval != nullptr) ? getauxval(AT_HWCAP) : 0;
+  unsigned long hwcap =
+#if HAVE_STRONG_GETAUXVAL
+      // Some compilers warn on (&getauxval != nullptr) in the block below.
+      getauxval(AT_HWCAP);
+#else
+      (&getauxval != nullptr) ? getauxval(AT_HWCAP) : 0;
```

Comment:
Won't this still issue a warning if built with an old compiler where HAVE_STRONG_GETAUXVAL is not defined?

Reply:
This is inside a block with the condition `defined (__linux__) && (HAVE_STRONG_GETAUXVAL || HAVE_WEAK_GETAUXVAL)`. So HAVE_WEAK_GETAUXVAL is supposed to be defined. Let me make that explicit...